### ISSUE TITLE
Empty submenus in elementary OS 5.0 Juno

### ISSUE THREAD

**Issue by kalligator** [ebruck/radiotray-ng#105]
Hello on eOS 5.0 Juno (18.04 amd64 base) I get empty submenus for everything (stations/prefs)

![radiotray-ng_0.2.4_amd64.deb](https://i.imgur.com/sfXSMpB.png[/img)
Stations placed in the root folder as well as Sleep Timer, About, Quit open normally.
I tried both the v0.2.3 provided .deb as well as building from the latest source (9 March 2019), same thing. 
I also tested other themes than the default. I even tried it on a live session just in case something was wrong with my setup, same deal. 

As for eOS 0.4.1 Loki the provided radiotray-ng_0.2.3_ubuntu_16.04_amd64.deb cannot even get installed, 
> Subprocess installed post-installation script returned error exit status 1 

Building from source is also not possible as the dependancies cannot be installed.

**Comment by ebruck** [comment]
It seems that there may be a problem with eOS 5.0's appindicator support. Can you try another application that also uses the tray to see if it has a problem?

**Comment by ebruck** [comment]
@kalligator is this how you turned on appindicator support?  If not then please give it a try.

https://www.linuxuprising.com/2018/08/how-to-re-enable-ayatana-appindicators.html

**Comment by kalligator** [comment]
Thanks for getting back, most probably that's where the problem lies in yes. 
I have followed this guide: https://github.com/mdh34/elementary-indicators - they probably make the same alterations but I'll give the one you mentioned a go just in case and I'll get back with before and after debug info by tomorrow.

Re the behaviour of other indicator apps, generally there are no obvious problems with Megasync, Viber. 
Radiotray-lite works ok but original radiotray has the same problem as radiotray-ng with submenus.

Also I forgot to mention, custom station images don't show (blank). If not set, the built-in shows normally.

**Comment by kalligator** [comment]
Nothing usefull I'm afraid:
```
[2019-03-11 01:30:02.754471] [0x00007f2db339dac0] [info ] (main.cpp:230) - radiotray-ng (v0.2.4-unknown) starting up
[2019-03-11 01:30:02.754621] [0x00007f2db339dac0] [debug] (config.cpp:29) - loading: /home/user/.config/radiotray-ng/radiotray-ng.json
[2019-03-11 01:30:03.004275] [0x00007f2db339dac0] [info ] (main.cpp:84) - debug logging enabled
[2019-03-11 01:30:03.004332] [0x00007f2db339dac0] [info ] (player.cpp:31) - starting gstreamer
[2019-03-11 01:30:06.411138] [0x00007f2db339dac0] [debug] (bookmarks.cpp:31) - loading: /home/user/.config/radiotray-ng/bookmarks.json
[2019-03-11 01:30:07.104622] [0x00007f2db339dac0] [debug] (file_monitor.hpp:108) - monitoring: /home/user/.config/radiotray-ng/bookmarks.json
[2019-03-11 01:30:07.105550] [0x00007f2db339dac0] [info ] (media_keys.cpp:90) - mapping volume up/down to: , 
[2019-03-11 01:30:07.105604] [0x00007f2db339dac0] [info ] (media_keys.cpp:93) - mapping station previous/next to: Previous, Next
[2019-03-11 01:30:07.105616] [0x00007f2db339dac0] [debug] (media_keys.cpp:184) - starting media keys
[2019-03-11 01:33:29.578360] [0x00007f2db339dac0] [info ] (radiotray_ng.cpp:457) - downloading: root, Poolside FM, http://stream.radio.co/s98f81d47e/listen
[2019-03-11 01:33:29.582221] [0x00007f2db339dac0] [debug] (playlist_downloader.cpp:132) - http-timeout=15
[2019-03-11 01:33:30.220905] [0x00007f2db339dac0] [warning] (playlist_downloader.cpp:191) - Content exceeds our limit of 4096 bytes!
[2019-03-11 01:33:30.221171] [0x00007f2db339dac0] [info ] (playlist_downloader.cpp:95) - no decoders, assuming direct media stream of content-type: audio/mpeg
[2019-03-11 01:33:30.221298] [0x00007f2db339dac0] [debug] (player.cpp:51) - uri: http://stream.radio.co/s98f81d47e/listen
[2019-03-11 01:33:30.221388] [0x00007f2db339dac0] [debug] (player.cpp:63) - buffer-size=640000, buffer-duration=2
[2019-03-11 01:33:30.234009] [0x00007f2db339dac0] [debug] (config.cpp:55) - saving: /home/user/.config/radiotray-ng/radiotray-ng.json
[2019-03-11 01:33:30.670559] [0x00007f2db339dac0] [debug] (player.cpp:253) - started buffering, setting state to: GST_STATE_PAUSED
[2019-03-11 01:33:32.702998] [0x00007f2db339dac0] [debug] (player.cpp:244) - stopped buffering, setting state to: GST_STATE_PLAYING
[2019-03-11 01:33:32.703468] [0x00007f2db339dac0] [debug] (notification.cpp:63) - notify: Poolside FM, Radiotray-NG, radiotray-ng-notification
[2019-03-11 01:33:34.008433] [0x00007f2db339dac0] [debug] (notification.cpp:63) - notify: The Best I Can, Basement Love, radiotray-ng-notification
[2019-03-11 01:33:42.353150] [0x00007f2db339dac0] [info ] (main.cpp:271) - radiotray-ng shutting down
[2019-03-11 01:33:42.353279] [0x00007f2db339dac0] [debug] (media_keys.cpp:218) - stopping media keys
[2019-03-11 01:33:42.369076] [0x00007f2db339dac0] [debug] (config.cpp:55) - saving: /home/user/.config/radiotray-ng/radiotray-ng.json
[2019-03-11 01:33:42.369779] [0x00007f2db339dac0] [info ] (player.cpp:39) - stopping gstreamer
```
It's not that big of a deal breaker anyway, I wouldn't use subfolders anyway, 10 or so select stations at the root folder are more than I need realistically.

**Comment by ebruck** [comment]
That and the fact the old radiotray & radiotray-lite have the same problem indicates that it's an eOS support of appindicator issue. Can you file a bug upstream?

**Comment by ebruck** [comment]
@kalligator I was playing around with Elementary OS 5.1 today and stumbled on a workaround for the submenus not being displayed. After Radiotray-NG loads, open up a terminal and run this command.

dbus-send --session --print-reply=literal --type=method_call --dest=com.github.radiotray_ng /com/github/radiotray_ng com.github.radiotray_ng.reload_bookmarks

**Comment by kalligator** [comment]
Nice catch! Out of learning curiosity, where did you spot what was needed? 

There's been a few days where I discovered an [updated version](https://github.com/Lafydev/wingpanel-indicator-ayatana) of the wingpanel-indicator-ayatana package that included among other commits fixes for the submenu issue globally. So all apps work properly. 

One minor annoyance is that there's an arbitrary cap of the icons size at 18px and it's solved as discussed [here](https://github.com/eth-p/wingpanel-indicator-ayatana/issues/9) on a previous iteration of the package (unmaintained). 

I tried making a .deb of the [final fixed package](https://github.com/kalligator/wingpanel-indicator-ayatana) but I'm out of my waters, couldn't handle the errors I was getting from packaging.  
My goal was to come back here with an easy to install .deb solution for all eOS users but couldn't make it :)
